sys/kern/subr_intr.c

```diff
@@ -100,12 +110,13 @@ MALLOC_DECLARE(M_INTRNG);
 MALLOC_DEFINE(M_INTRNG, "intr", "intr interrupt handling");
 
 /* Main interrupt handler called from assembler -> 'hidden' for C code. */
+void intr_irq_handler_root(struct trapframe *tf, uint32_t root);
 void intr_irq_handler(struct trapframe *tf);
```

Comment:
Ideally I'd actually like to get rid of `intr_irq_handler()` _now_.

Reply:
Just to clarify you mean replace all calls to `intr_irq_handler` with `intr_irq_handler_root(INTR_ROOT_IRQ)`? I kept the former to minimize the changes needed for archs that don't use multiple roots yet but I don't have a strong opinion on this so I can do either.

Reply:
Note, I am **not** a member of the organization so **do not** consider my opinions the last word.

My concern is compatibility shims tend to hang around a **long** time in FreeBSD.  There are likely plenty of decade-old fixup shims around.  This seems one which might be best to rip the band-aid off now, rather than hanging around for years.

Reply:
Sorry I should've added a new comment rather than edit my last one. I don't fully consider `intr_irq_handler` a compatibility shim since I expect some archs won't ever need the new version (e.g. ARM32 where the existing FIQ support was recently removed). That's why I'm waiting to see if anyone has opinions on this instead of just making the change you suggested.